src/main/scala/mesosphere/marathon/api/GroupApiService.scala

```diff
@@ -0,0 +1,71 @@
+package mesosphere.marathon
+package api
+
+import mesosphere.marathon.core.group.GroupManager
+import mesosphere.marathon.plugin.auth._
+import mesosphere.marathon.raml.{ GroupConversion, Raml }
+import mesosphere.marathon.state._
+
+import scala.async.Async._
+import scala.concurrent.{ ExecutionContext, Future }
+
+class GroupApiService(groupManager: GroupManager)(implicit authorizer: Authorizer, executionContext: ExecutionContext) {
+
+  /**
+    * Encapsulates the group update logic that is following:
+    * - if version is set, group is reverted to that version
+    * - if scaleBy is set, group is scaled up
+    * - if none version nor scaleBy is set then the group is updated
+    */
+  @SuppressWarnings(Array("all")) // async/await
+  def getUpdatedGroup(
+    rootGroup: RootGroup,
+    groupId: PathId,
+    groupUpdate: raml.GroupUpdate,
```

Comment:
What do you thing about separating raml and domain objects?

Reply:
that would be nice, but it would require bigger refactoring because some code inside those methods actually expect these raml types. Btw. this is not really a domain service, I intentionally placed it to the api package, so that it is very close to API.